lib/logstash/outputs/elasticsearch.rb

```diff
@@ -279,6 +279,8 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   def initialize(*params)
     super
     setup_ecs_compatibility_related_defaults
+
+    raise LogStash::ConfigurationError("`template_api` is not allowed unless a `template` is also provided") if original_params.include?('template_api') && !original_params.include?('template')
```

Comment:
Very very small breaking risk. The intent of `template_api` was _always_ as a hint of the template that was being provided, and never _successfully_ controlled which API was used for the default template. It is possible that a user explicitly added a meaningless `template_api` hint without a `template` that _happened_ to align with the default behaviour that _did_ work, but even these cases we should reject the config helpfully.